### ISSUE TITLE
Feature request: make rich text image chooser alt text field optional

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#6494]
### Is your proposal related to a problem?

In rich text, users get the opportunity to customize alt text when inserting an image. This is great, except the field should be switched from required to optional. Images not having alt text is a valid use case.

![wagtail-alt-chooser](https://user-images.githubusercontent.com/877585/97051659-739ea380-1577-11eb-8c77-4f80d39c9622.png)

### Describe the solution you'd like

The field should be optional. The idea solution would be to have a UI similar to TinyMCE’s image chooser:

![tinymce-alt-chooser](https://user-images.githubusercontent.com/877585/97051698-89ac6400-1577-11eb-978f-ace0060fef57.png)

This adds client-side validation to force end users to make a conscious choice that the image is decorative (thus no alt text), rather than risking they leave the field empty in error.

### Additional context

Corresponding Django form:

https://github.com/wagtail/wagtail/blob/516150403b89aaeb3a50667e6acffcb7472b66b3/wagtail/images/forms.py#L59-L68

Template:

https://github.com/wagtail/wagtail/blob/ba6f94def17b8bbc66002cbc7af60ed422658ff1/wagtail/images/templates/wagtailimages/chooser/select_format.html#L13-L18

Note Wagtail currently also has default alt text in this field – this is a separate issue. If we keep the default alt text, we need to make sure that choosing "image is decorative" subsequently clears the alt text.

**Comment by thibaudcolas** [comment]
@helenb and I are going to be working on this some time over the next few months. Our intention is to add this to rich text image choosers with the "ideal" implementation suggested above, forcing end users to make an explicit choice. 

We’re thinking of implementing this "image is decorative" field client-side only, with no corresponding Django model, because its only purpose is to hide the (then optional) alt text field if checked. Also happy to do it differently if others have suggestions.

**Comment by Scotchester** [comment]
I like this approach! I wonder if the is-decorative checkbox should also empty the field out, but I'm wary of clearing users' work without warning.

**Comment by thibaudcolas** [comment]
Discussed with Scott – if "decorative" is checked, we’ll want to ignore the contents of the alt text field on save, but not discard it whenever checking/unchecking.
